app/Http/Controllers/AjaxController.php

```diff
@@ -29,4 +29,31 @@ public function addfriend(Request $req) {
         }
         return response()->json(array("message" => "generic error message"), 500);
     }
+
+    /*
+     * AJAX Request to handle accepting/declining a friend request
+     *
+     */
+    public function processFriendRequest(Request $req) {
+        if ($req->json() && isset ($req->username)) {
+            $friendship = Friend::where('username2', Auth::user()->username)
+                ->where('username1', $req->username)
+                ->where('status_code', 'pending')
+                ->where('action_username', '!=', Auth::user()->username)
+                ->get();
+            if (isset ($req->accepted)) {
+                if ($req->accepted == 'accepted') {
+                    $friendship->status_code = 'accepted';
+                    $friendship->save();
+                    return response()->json(array("message" => "friend request accepted!"), 200);
+                } else if ($req->accepted == 'denied') {
+                    $friendship->status_code = 'denied';
+                    $friendship->save();
+                    return response()->json(array("message" => "friend request denied!"), 200);
```

Comment:
@nikoootine @vincent-h-lee what convention do we want to do for our AJAX. 200 means ok, but do we want to send back an Error code if the request is denied?